featuretools_tsfresh_primitives/utils.py

```diff
@@ -1,45 +1,55 @@
-import numpy as np
-import pandas as pd
+from featuretools.primitives import AggregationPrimitive, TransformPrimitive
+from tsfresh.feature_extraction.settings import ComprehensiveFCParameters
 
 import featuretools_tsfresh_primitives
 
 
-def _pascal_case(snake_str):
-    words = snake_str.split('_')
-    titled_words = map(str.title, words)
-    return ''.join(titled_words)
+def supported_primitives():
+    """Generates the currently supported primitives.
 
+        Returns:
+            generator: primitive classes
+    """
+    types = AggregationPrimitive, TransformPrimitive
+    for key in dir(featuretools_tsfresh_primitives):
+        value = getattr(featuretools_tsfresh_primitives, key)
+        is_primitive = isinstance(value, type) and issubclass(value, types)
+        if is_primitive: yield value
 
-def _tsfresh_to_featuretools(fc_name):
-    return getattr(featuretools_tsfresh_primitives, _pascal_case(fc_name))
 
+def comprehensive_primitives(fc_parameters=None):
+    """Creates a mapping for primitive instances from a dictionary list of parameters.
 
-def primitives_from_fc_settings(fc_settings):
-    """
-    Return a list of :class:AggregationPrimitive from tsfresh settings.
-    The format is the same as the argument `default_fc_parameters` of
-    :func:`tsfresh.feature_extraction.extract_features`
+        Args:
+            fc_parameters (ComprehensiveFCParameters): An instance of :class:`ComprehensiveFCParameters`
```

Comment:
We should have a test specifically for passing a value for `fc_parameters`

Reply:
What you think about a docstring test for this example?
https://github.com/FeatureLabs/featuretools-tsfresh-primitives/pull/48/files#diff-3a0dc45f9b8cc21ed8179728be9743cfR31-R35

Reply:
Oh I see, it's tested in the docstring